[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39410 [vulkan] Remove unused variables
* #39384 [vulkan] Shaders caching
* #39282 [vulkan] jit passes for vulkan conv2 prepack and fuse with clamp
* #39115 [vulkan] Conv2d with optional clamp
* #39078 [vulkan] addmm support non-vulkan inputs
* #39077 [vulkan] VulkanTensor, add strides in interface
* **#39076 [vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend**

`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

Differential Revision: [D21962428](https://our.internmc.facebook.com/intern/diff/D21962428)

IvanKobzarev committed:
[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend

[ghstack-poisoned]

IvanKobzarev committed:
drop batch norm stub as fuse_models Conv2-BatchNorm2 works on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

[ghstack-poisoned]

IvanKobzarev committed:
rebase on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

[ghstack-poisoned]

AshkanAliabadi commented on binaries/speed_benchmark_torch.cc:
```diff
@@ -136,9 +140,15 @@ std::vector<c10::IValue> create_inputs() {
     }
 
-    inputs.push_back(torch::ones(
+    auto input_tensor = torch::ones(
```
Minor suggestion: consider marking this as const.  const is not implied by auto.

AshkanAliabadi commented on binaries/speed_benchmark_torch.cc:
```diff
@@ -136,9 +140,15 @@ std::vector<c10::IValue> create_inputs() {
+        at::TensorOptions(input_type).memory_format(input_memory_format));
+    if (FLAGS_vulkan) {
+      auto input_tensor_vulkan = input_tensor.vulkan();
+      inputs.push_back(input_tensor_vulkan);
```
You could consider using ```inputs.push_back(input_tensor.vulkan())``` instead to take advantage of .push_back()'s overload of r-value references.

Both of these comments are very minor, but in case you are interested in some tips which you are more than welcome to ignore.

IvanKobzarev committed:
Update on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

Differential Revision: [D21962428](https://our.internmc.facebook.com/intern/diff/D21962428)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

Differential Revision: [D21962428](https://our.internmc.facebook.com/intern/diff/D21962428)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] speed_becnhmark_torch add vulkan arg to use Vulkan backend"


`--vulkan` argument to use torch benchmark on Vulkan Backend
if it is True - inputs will be converted to Vulkan backend before module.forward

Usage for mobilenetv2 fp32:
```
/build/bin/speed_benchmark_torch --model=mn-fp32.pt --input_type=float --input_dims=1,3,224,224 --warmup=1 --iter=5 --vulkan=true
```

Differential Revision: [D21962428](https://our.internmc.facebook.com/intern/diff/D21962428)

[ghstack-poisoned]
